packages/salesforcedx-vscode-core/test/vscode-integration/commands/baseDeployRetrieve.test.ts

```diff
@@ -191,14 +194,28 @@ describe('Base Deploy Retrieve Commands', () => {
     it('should use the api version from SFDX configuration', async () => {
       const executor = new TestDeployRetrieve();
       const configApiVersion = '30.0';
-      sb.stub(ConfigUtil, 'getConfigValue')
-        .withArgs('apiVersion')
-        .returns(configApiVersion);
+      sb.stub(ConfigUtil, 'getUserConfiguredApiVersion').resolves(
+        configApiVersion
+      );
+      const getOrgApiVersionSpy = sb.spy(OrgAuthInfo, 'getOrgApiVersion');
 
       await executor.run({ data: {}, type: 'CONTINUE' });
       const components = executor.lifecycle.doOperationStub.firstCall.args[0];
 
       expect(components.apiVersion).to.equal(configApiVersion);
+      expect(getOrgApiVersionSpy.called).to.be.false;
+    });
+
+    it('should use the api version from the Org when no User-configured api version is set', async () => {
+      const executor = new TestDeployRetrieve();
+      const orgApiVersion = '40.0';
+      sb.stub(ConfigUtil, 'getUserConfiguredApiVersion').resolves(undefined);
+      sb.stub(OrgAuthInfo, 'getOrgApiVersion').resolves(orgApiVersion);
+
+      await executor.run({ data: {}, type: 'CONTINUE' });
+      const components = executor.lifecycle.doOperationStub.firstCall.args[0];
+
```

Comment:
nit city: in unit tests where I'm stubbing/spying I always like to ensure that the all stubs were called as expected.  So you could add a expects to verify each stub was called once here.